integration/network/service_test.go

```diff
@@ -9,10 +9,194 @@ import (
+	assert.Equal(t, out.IPAM.Config[0].Subnet, "172.18.0.0/16")
+}
+
+func TestDaemonDefaultNetworkPools(t *testing.T) {
+	// Remove docker0 bridge and the start daemon defining the predefined address pools
+	defaultNetworkBridge := "docker0"
+	delInterface(t, defaultNetworkBridge)
+	d := daemon.New(t)
+	defer d.Stop(t)
+	d.Start(t,
+		"--default-node-local-address-pools", "base=175.30.0.0/16,size=16",
+		"--default-node-local-address-pools", "base=175.33.0.0/16,size=24")
+
+	// Verify bridge network's subnet
+	cli, err := d.NewClient()
+	assert.Assert(t, err)
+	defer cli.Close()
+	out, err := cli.NetworkInspect(context.Background(), "bridge", types.NetworkInspectOptions{})
+	assert.NilError(t, err)
+	assert.Equal(t, out.IPAM.Config[0].Subnet, "175.30.0.0/16")
+
+	// Create a bridge network and verify its subnet is the second default pool
+	name := "elango"
+	networkCreate := types.NetworkCreate{
+		CheckDuplicate: false,
+	}
+	networkCreate.Driver = "bridge"
+	_, err = cli.NetworkCreate(context.Background(), name, networkCreate)
+	assert.NilError(t, err)
+	out, err = cli.NetworkInspect(context.Background(), name, types.NetworkInspectOptions{})
+	assert.NilError(t, err)
+	assert.Equal(t, out.IPAM.Config[0].Subnet, "175.33.0.0/24")
+
+	// Create a bridge network and verify its subnet is the third default pool
+	name = "saanvi"
+	networkCreate = types.NetworkCreate{
+		CheckDuplicate: false,
+	}
+	networkCreate.Driver = "bridge"
+	_, err = cli.NetworkCreate(context.Background(), name, networkCreate)
+	assert.NilError(t, err)
+	out, err = cli.NetworkInspect(context.Background(), name, types.NetworkInspectOptions{})
+	assert.NilError(t, err)
+	assert.Equal(t, out.IPAM.Config[0].Subnet, "175.33.1.0/24")
+	delInterface(t, defaultNetworkBridge)
+
+}
+
+func TestDaemonRestartWithExistingNetwork(t *testing.T) {
+	defaultNetworkBridge := "docker0"
+	d := daemon.New(t)
+	d.Start(t)
+	defer d.Stop(t)
+	// Verify bridge network's subnet
+	cli, err := d.NewClient()
+	assert.Assert(t, err)
+	defer cli.Close()
+
+	// Create a bridge network
+	name := "elango"
+	networkCreate := types.NetworkCreate{
+		CheckDuplicate: false,
+	}
+	networkCreate.Driver = "bridge"
+	_, err = cli.NetworkCreate(context.Background(), name, networkCreate)
+	assert.NilError(t, err)
+	out, err := cli.NetworkInspect(context.Background(), name, types.NetworkInspectOptions{})
+	assert.NilError(t, err)
+	networkip := out.IPAM.Config[0].Subnet
+
+	// Restart daemon with default address pool option
+	d.Restart(t,
+		"--default-node-local-address-pools", "base=175.30.0.0/16,size=16",
+		"--default-node-local-address-pools", "base=175.33.0.0/16,size=24")
+
+	out1, err := cli.NetworkInspect(context.Background(), name, types.NetworkInspectOptions{})
+	assert.NilError(t, err)
+	assert.Equal(t, out1.IPAM.Config[0].Subnet, networkip)
+	delInterface(t, defaultNetworkBridge)
+}
+
+func isIPAllocationValid(ip1 string, ip2 string) bool {
```

Comment:
This seems not needed.

Reply:
The reason I added is, in case if we want to do more additional check with IP address range down the line , I can expand this API. 
We will have to expand this feature for global scope in near future.

Reply:
We can always add a helper later, right now it's  not helping anything.